packages/core/src/browser/messaging/ws-connection-provider.ts

```diff
@@ -45,7 +51,11 @@ export class WebSocketConnectionProvider extends AbstractConnectionProvider<WebS
         const url = this.createWebSocketUrl(WebSocketChannel.wsPath);
         const socket = this.createWebSocket(url);
         socket.onerror = console.error;
+        socket.onopen = () => {
+            this.onSocketOpenedEmitter.fire(undefined);
+        };
         socket.onclose = ({ code, reason }) => {
+            this.onSocketClosedEmitter.fire(undefined);
```

Comment:
Should be called after calling `channel.close(code, reason);` on each channel, not before.

Reply:
Right, good catch, thanks! Fixed.